step-release-vis/src/app/components/environment/environment.ts

```diff
@@ -194,19 +196,27 @@ export class EnvironmentComponent implements OnInit, OnChanges {
   }
 
   moveTooltip(event: MouseEvent): void {
-    this.tooltip.mouseX = event.pageX - window.scrollX;
-    this.tooltip.mouseY = event.pageY - window.scrollY;
-    this.tooltip.show = true;
+    if (!this.tooltip.clickOn) {
+      this.tooltip.mouseX = event.pageX - window.scrollX;
+      this.tooltip.mouseY = event.pageY - window.scrollY;
+      this.tooltip.show = true;
 
-    const svgElement = document.getElementById(this.environment.name + '-svg');
-    const svgMouseX = event.pageX - svgElement.getBoundingClientRect().left;
+      const svgElement = document.getElementById(
+        this.environment.name + '-svg'
+      );
+      const svgMouseX = event.pageX - svgElement.getBoundingClientRect().left;
 
-    this.updateCurrentSnapshot(svgMouseX);
+      this.updateCurrentSnapshot(svgMouseX);
+    }
   }
 
-  leftEnvironment(event: MouseEvent): void {
-    this.hideTooltip();
-    this.currentSnapshot = undefined;
+  // TODO(ancar): Make it work for when it leaves the environment, not
```

Comment:
this should be like  "if not in tooltoip >> hide"
and on the leave tooltip "if not in env >> hide" the clickon is implicit from the in tooltip condition, as if it would not be clicked you can not get inside.

Reply:
forgot to delete the TODO

Reply:
ok, will try and change

Reply:
the problem is I cannot really check in the leftEnvironment if the tooltip is hovered, because when it enters the tooltip it considers that the environment is left and after that it gives me the emitted value that the tooltip is hovered.

Reply:
the problem is I cannot check if the tooltip is hovered in the leftEnvironment function as I get the value true emitted from the child that the tooltip is hovered after the environment is left